zhaquirks/tuya/electric_heating.py

```diff
@@ -0,0 +1,167 @@
+                value * 100,  # degree to centidegree
+            )
+        elif attrid == MOESBHT_TEMPERATURE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event(
+                "temperature_change",
+                "local_temp",
+                value * 10,  # decidegree to centidegree
+            )
+        elif attrid == MOESBHT_SCHEDULE_MODE_ATTR:
+            if value == 0:  # value is inverted
+                self.endpoint.device.thermostat_bus.listener_event(
+                    "program_change", "scheduled"
+                )
+        elif attrid == MOESBHT_MANUAL_MODE_ATTR:
+            if value == 0:  # value is inverted
+                self.endpoint.device.thermostat_bus.listener_event(
+                    "program_change", "manual"
+                )
+        elif attrid == MOESBHT_ENABLED_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event("enabled_change", value)
+        elif attrid == MOESBHT_RUNNING_MODE_ATTR:
+            self.endpoint.device.thermostat_bus.listener_event("state_change", value)
+        elif attrid == MOESBHT_CHILD_LOCK_ATTR:
+            self.endpoint.device.ui_bus.listener_event("child_lock_change", value)
+
+
+class MoesBHTThermostat(TuyaThermostatCluster):
+    """Thermostat cluster for some electric heating controllers."""
+
+    def map_attribute(self, attribute, value):
+        """Map standardized attribute value to dict of manufacturer values."""
+
+        if attribute == "occupied_heating_setpoint":
+            # centidegree to degree
+            return {MOESBHT_TARGET_TEMP_ATTR: round(value / 100)}
+        if attribute == "system_mode":
+            if value == self.SystemMode.Off:
+                return {MOESBHT_ENABLED_ATTR: 0}
+            if value == self.SystemMode.Heat:
+                return {MOESBHT_ENABLED_ATTR: 1}
+            self.error("Unsupported value for SystemMode")
+        elif attribute == "programing_oper_mode":
+            # values are inverted
+            if value == self.ProgrammingOperationMode.Simple:
+                return {MOESBHT_MANUAL_MODE_ATTR: 0, MOESBHT_SCHEDULE_MODE_ATTR: 1}
+            if value == self.ProgrammingOperationMode.Schedule_programming_mode:
+                return {MOESBHT_MANUAL_MODE_ATTR: 1, MOESBHT_SCHEDULE_MODE_ATTR: 0}
+            self.error("Unsupported value for ProgrammingOperationMode")
+
+        return super().map_attribute(attribute, value)
+
+    def program_change(self, mode):
+        """Programming mode change."""
+        if mode == "manual":
+            value = self.ProgrammingOperationMode.Simple
+        else:
+            value = self.ProgrammingOperationMode.Schedule_programming_mode
+
+        self._update_attribute(self.attridx["programing_oper_mode"], value)
+
+    def enabled_change(self, value):
+        """System mode change."""
+        if value == 0:
+            mode = self.SystemMode.Off
+        else:
+            mode = self.SystemMode.Auto
```

Comment:
Changing this line to:
`mode = self.SystemMode.Heat`
Fixed all the problems. Now it's showing correct value for target temp. Changing the value on the thermostat updates the value on home-assistant and viceversa.

Reply:
Sorry, I missed that occurrence of `Auto` when I replaced them.